test/Microsoft.AspNetCore.SignalR.Tests/HubConnectionHandlerTests.cs

```diff
@@ -148,6 +148,58 @@ public class HubConnectionHandlerTests
             }
         }
 
+        [Fact]
+        public async Task OberserverDoesntThrowWhenOnNextIsCalledAfterChannelIsCompleted()
+        {
+            var observable = new Observable<int>();
+            var serviceProvider = HubConnectionHandlerTestUtils.CreateServiceProvider(s => s.AddSingleton(observable));
+            var connectionHandler = serviceProvider.GetService<HubConnectionHandler<ObservableHub>>();
+
+            var waitForSubscribe = new TaskCompletionSource<object>();
+            observable.OnSubscribe = o =>
+            {
+                waitForSubscribe.TrySetResult(null);
+            };
+
+            var waitForDispose = new TaskCompletionSource<object>();
+            observable.OnDispose = o =>
+            {
+                waitForDispose.TrySetResult(null);
+            };
+
+            using (var client = new TestClient())
+            {
+                var connectionHandlerTask = await client.ConnectAsync(connectionHandler);
+
+                async Task Subscribe()
+                {
+                    var results = await client.StreamAsync(nameof(ObservableHub.Subscribe));
+                }
+
+                var subscribeTask = Subscribe();
```

Comment:
This can just be `var subscribeTask = client.StreamAsync(nameof(ObservableHub.Subscribe))`. We can still `await` it below and just discard the result without needing the local function.